### PR TITLE
Remove code coverage by directory view and backend API

### LINKED ISSUE #837: shipit_frontend, shipit_uplift: Remove coverage by directory view and API
It is no longer used.

### PR DESCRIPTION
**Pull request by Sush0512**
Fixes #837.

### CODE REVIEW

File: src/shipit_uplift/shipit_uplift/api.py [hunk 1]
```diff
@@ -21,7 +21,7 @@
 )
 from shipit_uplift.config import SCOPES_USER, SCOPES_BOT, SCOPES_ADMIN
 from shipit_uplift import (
-    coverage_by_dir_impl, coverage_for_file_impl,
+     coverage_for_file_impl,
```
Comment (marco-c) [inline src/shipit_uplift/shipit_uplift/api.py]: There's an additional whitespace here that shouldn't be there

File: src/shipit_uplift/shipit_uplift/api.py [hunk 2]
```diff
@@ -383,8 +383,8 @@ def create_patch_status(bugzilla_id):
     return serialize_patch_status(ps)
 
 
-def coverage_by_dir(path=''):
-    return coverage_by_dir_impl.generate(path)
+#def coverage_by_dir(path=''):
```
Comment (marco-c) [inline src/shipit_uplift/shipit_uplift/api.py]: Let's remove it altogether, not just comment it out

File: src/shipit_uplift/shipit_uplift/api.yml [hunk 3]
```diff
@@ -159,18 +159,7 @@ paths:
...
-      parameters:
-      - name: path
-        in: query
-        description: Path
-        required: false
-        type: string
-      responses:
-        200:
-          description: Code coverage information by directory
+
```
Comment (marco-c) [inline src/shipit_uplift/shipit_uplift/api.yml]: Nit: Remove this additional newline

**Comment by marco-c** [comment]
You can run the tests locally with `./please check shipit-uplift` and `./please check shipit-frontend`.

**Comment by Sush0512** [comment]
Hi @marco-c,
Please find the following updates
- removed extra spaces, lines
- synced the fork
- squashed the extra commits
- run the tests for shipit_uplift.

for shipit-frontend, it is throwing me an error saying there are no checks found for shipit-frontend.

**Comment by marco-c** [comment]
> squashed the extra commits

There are several unrelated changes in the PR, can you try rebasing on master again?

**Comment by Sush0512** [comment]
@marco-c fixed it. Thank you :+1:

File: src/shipit_frontend/src/App/CodeCoverage.elm [hunk 4]
```diff
@@ -1,249 +0,0 @@
-module App.CodeCoverage exposing (..)
```
Comment (marco-c) [inline src/shipit_frontend/src/App/CodeCoverage.elm]: There's probably some code under src/shipit_frontend that is referencing this that should be removed too.

**Comment by marco-c** [comment]
There's still something that should be removed, see the build logs.

Have you tried running `./please run shipit-frontend`?

**Comment by Sush0512** [comment]
Hi @marco-c,
Sorry for the late reply. I had an issue with the run command and fixed it recently. 
I fixed the error causing snippet 
```
div [ class "collapse navbar-collapse" ]
        [ ul [ class "navbar-nav mr-auto " ]
            (viewNavDashboard model
                ++ [ li [ class "nav-item" ]
                        [ eventLink (ShowCodeCoverage Nothing) [ class "nav-link" ] [ text "Code Coverage" ]
                        ]
                   ]
            )
        , ul [ class "navbar-nav" ] (viewUser model)
]
```
by making the second parameter of `div` function empty, i.e, `div [ class "collapse navbar-collapse" ][]`. Is this fine? Now the build is successful.

File: src/shipit_frontend/src/App.elm [hunk 5]
```diff
@@ -265,7 +234,7 @@ viewNavBar model =
         [ ul [ class "navbar-nav mr-auto " ]
```
Comment (marco-c) [inline src/shipit_frontend/src/App.elm]: We need to remove this "Code Coverage" ul completely.

File: src/shipit_frontend/src/App.elm [hunk 6]
```diff
@@ -261,16 +230,7 @@ viewNavBar model =
...
         [ text "Uplift Dashboard" ]
-    , div [ class "collapse navbar-collapse" ]
-        [ ul [ class "navbar-nav mr-auto " ]
-            (viewNavDashboard model
-                ++ [ li [ class "nav-item" ]
-                        [ eventLink (ShowCodeCoverage Nothing) [ class "nav-link" ] [ text "Code Coverage" ]
-                        ]
-                   ]
-            )
-        , ul [ class "navbar-nav" ] (viewUser model)
```
Comment (marco-c) [inline src/shipit_frontend/src/App.elm]: Only the "Code Coverage" one should be removed, not also this ul.

File: src/shipit_frontend/src/App.elm [hunk 7]
```diff
@@ -505,7 +455,8 @@ delta2url previous current =
                                 [ "release-dashboard", toString analysis.id ]
 
                             _ ->
-                                [ "release-dashboard" ]
+                                [ "release-dashboard" ]
```
Comment (marco-c) [inline src/shipit_frontend/src/App.elm]: Please remove unrelated changes here

File: src/shipit_frontend/src/App.elm [hunk 8]
```diff
@@ -516,26 +467,12 @@ delta2url previous current =
...
-                                []
-                in
-                Maybe.map
-                    (Builder.prependToPath ([ "code-coverage" ] ++ parts))
-                    (Just builder)
-
             _ ->
                 Maybe.map
                     (Builder.prependToPath [])
-                    (Just builder)
```
Comment (marco-c) [inline src/shipit_frontend/src/App.elm]: Please remove unrelated changes here too

**Comment by Sush0512** [comment]
Carefully checked and updated all the files.
All the checks pass.

**Comment by marco-c** [comment]
The build is still failing while validating the elm file.
You can run the check locally using `./please shell shipit-frontend` and then `elm-format-0.18 src/ --validate`.

**Comment by Sush0512** [comment]
Hi @marco-c 
I have run the commands you mentioned.
shipit-frontend
The build was successful but when tried to run the command `elm-format-0.18 src/ --validate.`  It throws an error saying please check the `elm-format-0.18` 
Can you please provide some insights on how to fix it, since there was no code added in the files.

**Comment by garbas** [comment]
@Sush0512 we require the elm code to be formated with [elm-format](https://github.com/avh4/elm-format) tool. What you need to do is:

- enter the development environment, where this tool is available (`./please shell shipit-frontend`)
- format the failing file (`$ elm-format-0.18 src/App.elm`), you will have to press `y`
- commit the change

**Comment by marco-c** [comment]
@Sush0512 need any help? Is what @garbas said clear?

**Comment by Sush0512** [comment]
Yes. Sorry for addressing late. Fixes #952

**Comment by marco-c** [comment]
> Yes. Sorry for addressing late.

Don't worry, there's no rush! 😄

**Comment by Sush0512** [comment]
Thanks! :)
